messaging/src/main/java/io/lantern/messaging/metadata/Metadata.kt

```diff
@@ -279,23 +290,47 @@ class Metadata(
             codec.stop()
             codec.release()
             extractor.release()
-            val floats = FloatArray(BAR_COUNT)
-            val ints = IntArray(BAR_COUNT)
+
+            // When we skip data on the read end, the presentationTimeUs on decoded end is sometimes
+            // incorrect. It's always correctly sequenced, but the timeline may appear much shorter.
+            // So above, we just collected the individual samples and corresponding timestamps, and
+            // here we scale them back to the actual duration of the file.
+            val wave = LongArray(numberOfBars)
```

Comment:
This was another interesting thing I noticed. We track position in the audio stream based on the `presentationTimeUs`. This is a value that's passed in on the input buffer side, and it's also made available on the output buffer side. I had assumed that the value would match, but it turns out that the value supplied to the input buffer is just a hint and the output buffer for MP3 seems to actually set the time based on number of samples it has decoded. When we don't skip any samples, that's not a problem, but when we do skip samples, it ends up looking like we have data for the first few bars of the waveform and then nothing.

So, what I do is record the total duration as seen on the output buffer side, then scale that up to what we know is the real duration of the audio file. Only after that point do I generate the bars.